sdk/core/core/src/az_span.c

```diff
@@ -643,39 +643,77 @@ AZ_NODISCARD az_result _az_span_url_encode(az_span destination, az_span source,
   int32_t const source_size = az_span_size(source);
   _az_PRECONDITION_VALID_SPAN(destination, source_size, false);
 
-  int32_t result_size = 0;
-  for (int32_t i = 0; i < source_size; ++i)
+  if (source_size == 0)
```

Comment:
As far as I can tell, this check isn't necessary and removing it wouldn't change the behavior when `source_size == 0`. It should just fall down to the end and return `AZ_OK` with `*out_length = 0`.

Reply:
I'd prefer to keep it this way: I know that the code below would work correctly for source_size == 0, but it is a convenience to write it below knowing that I don't have to proofread every line for source being 0. And also if there's nothing to do, we exit rather fast, not doing the extra job, eating battery and so on. The whole check should be pretty cheap, since checking for zero are sometimes even cheaper than checking with any other value, something along the lines of "test EAX, EAX; jz $exit0".